File: src/demo_tf/scripts/spawn_turtle.py
#! /usr/bin/env python3

import rospy
from turtlesim.srv import Spawn, SpawnRequest, SpawnResponse

if __name__ == "__main__":

    rospy.init_node("turtle_spawn_p")

    client = rospy.ServiceProxy("/spawn",Spawn)

    client.wait_for_service()

    req = SpawnRequest()
    req.x = 1.0
    req.y = 1.0
    req.theta = 3.14
    req.name = "turtle2"

    try:
        response = client.call(req)
        rospy.loginfo("Success, name is: %s", response.name)
    except Exception as e:
        rospy.loginfo("Error...")

File: src/demo_tf/src/spawn_turtle.cpp
#include "ros/ros.h"
#include "turtlesim/Spawn.h"

int main(int argc, char *argv[])
{
    ros::init(argc,argv,"create_turtle");

    ros::NodeHandle nh;

    ros::ServiceClient client = nh.serviceClient<turtlesim::Spawn>("/spawn");

    ros::service::waitForService("/spawn");
    turtlesim::Spawn spawn;
    spawn.request.name = "turtle2";
    spawn.request.x = 1.0;
    spawn.request.y = 2.0;
    spawn.request.theta = 3.1415926;
    bool flag = client.call(spawn);
    if (flag)
    {
        ROS_INFO("Turtle %s Success!", spawn.response.name.c_str());
    }
    else
    {
        ROS_INFO("Failed!");
    }

    ros::spin();

    return 0;
}